src/dmd/expressionsem.d

```diff
@@ -4113,7 +4113,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         // Type is a "delegate to" or "pointer to" the function literal
         if ((exp.fd.isNested() && exp.fd.tok == TOK.delegate_) || (exp.tok == TOK.reserved && exp.fd.treq && exp.fd.treq.ty == Tdelegate))
         {
-            exp.type = new TypeDelegate(exp.fd.type);
+            exp.type = new TypeDelegate(exp.fd.type.toTypeFunction());
```

Comment:
change toTypeFunction to isTypeFunction. The latter will return null if it is not, and the next line will fault. Everything else uses the `is` functions, we shouldn't be created a same-only-different parallel set.